src/index.ts

```diff
@@ -6,21 +6,22 @@ import { init as initTezos } from "./tezos";
 import { warehouseHandlers } from "./handlers/warehouse";
 
 async function start() {
-    function shutdown(exitCode: number) {
-        drain();
+    async function shutdown(exitCode: number) {
+        await drain();
         process.exit(exitCode);
     }
 
     try {
         await initNats();
         await initTezos();
 
-        registerHandlers(warehouseHandlers);
+        await registerHandlers(warehouseHandlers);
```

Comment:
adding the `await` here ensures that the `try/catch` block will catch error thrown in the handlers when they're registered